pkg/trace/api/debugger.go

```diff
@@ -60,49 +78,95 @@ func debuggerErrorHandler(err error) http.Handler {
 }
 
 // newDebuggerProxy returns a new httputil.ReverseProxy proxying and augmenting requests with headers containing the tags.
-func newDebuggerProxy(conf *config.AgentConfig, target *url.URL, key string, tags string) *httputil.ReverseProxy {
+func newDebuggerProxy(conf *config.AgentConfig, targets []*url.URL, keys []string, tags string) *httputil.ReverseProxy {
 	logger := log.NewThrottled(5, 10*time.Second) // limit to 5 messages every 10 seconds
-	cidProvider := NewIDProvider(conf.ContainerProcRoot)
 	director := func(req *http.Request) {
-		ddtags := tags
-		containerID := cidProvider.GetContainerID(req.Context(), req.Header)
-		if ct := getContainerTags(conf.ContainerTags, containerID); ct != "" {
-			ddtags = fmt.Sprintf("%s,%s", ddtags, ct)
-		}
-		q := req.URL.Query()
-		if qtags := q.Get("ddtags"); qtags != "" {
-			ddtags = fmt.Sprintf("%s,%s", ddtags, qtags)
-		}
-		q.Set("ddtags", ddtags)
-		newTarget := *target
-		newTarget.RawQuery = q.Encode()
-		req.Header.Set("DD-API-KEY", key)
 		req.Header.Set("DD-REQUEST-ID", uuid.New().String())
 		req.Header.Set("DD-EVP-ORIGIN", "agent-debugger")
-		req.URL = &newTarget
-		req.Host = target.Host
 	}
 	return &httputil.ReverseProxy{
-		Director:  director,
-		ErrorLog:  stdlog.New(logger, "debugger.Proxy: ", 0),
-		Transport: &measuringDebuggerTransport{conf.NewHTTPTransport()},
+		Director: director,
+		ErrorLog: stdlog.New(logger, "debugger.Proxy: ", 0),
+		Transport: &measuringDebuggerMultiTransport{
+			conf.NewHTTPTransport(),
+			targets,
+			keys,
+			tags,
+			conf,
+			NewIDProvider(conf.ContainerProcRoot),
+		},
 	}
 }
 
-// measuringDebuggerTransport sends HTTP requests to a defined target url. It also sets the API keys in the headers.
-type measuringDebuggerTransport struct {
-	rt http.RoundTripper
+// measuringDebuggerMultiTransport sends HTTP requests to a defined target url. It also sets the API keys in the headers.
+type measuringDebuggerMultiTransport struct {
+	rt          http.RoundTripper
+	targets     []*url.URL
+	keys        []string
+	tags        string
+	conf        *config.AgentConfig
+	cidProvider IDProvider
 }
 
-func (m *measuringDebuggerTransport) RoundTrip(req *http.Request) (res *http.Response, err error) {
+func (m *measuringDebuggerMultiTransport) RoundTrip(req *http.Request) (rres *http.Response, rerr error) {
```

Comment:
The code is very similar to the [profiler round tripper](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/api/profiles.go), can we factorize the common code?

Reply:
Also applies to the loop that parses `AdditionalEndpoints`

Reply:
TBH, I'm not sure it is worth it. The code is not very complicated and it contains numerous references to different config settings, constants and functions. The extracted common functionality would be very small. I think it's not worth it to extract and create a functional dependency between two unrelated products for the sake of minimal DRYness. WDYT?

Reply:
The round tripper is almost a copy paste from profiling, there is good chunk of code that can be shared. I'm just thinking about the maintainability of the code here as a code owner. I also understand a refactor can be considered out of the scope of your task.

Reply:
I'll have a shot at it later. I'm going to add another endpoint that will be very similar for a new feature soon (hopefully before the freeze), so this might pay off. I'm still apprehensive to touch profiling code.

Reply:
I've put the refactoring in an extra commit: https://github.com/DataDog/datadog-agent/pull/16409/commits/ba2674e226a1f089d4ca579c99b7d255000e6f0e
Let me know what you think.